esi_leap/objects/owner_change.py

```diff
@@ -0,0 +1,185 @@
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import datetime
+from oslo_config import cfg
+from oslo_log import log as logging
+from oslo_versionedobjects import base as versioned_objects_base
+
+from esi_leap.common import exception
+from esi_leap.common import statuses
+from esi_leap.common import utils
+from esi_leap.db import api as dbapi
+from esi_leap.objects import base
+from esi_leap.objects import fields
+from esi_leap.objects import lease as lease_obj
+from esi_leap.objects import offer as offer_obj
+from esi_leap.resource_objects import resource_object_factory as ro_factory
+
+
+CONF = cfg.CONF
+LOG = logging.getLogger(__name__)
+
+
+@versioned_objects_base.VersionedObjectRegistry.register
+class OwnerChange(base.ESILEAPObject):
+    dbapi = dbapi.get_instance()
+
+    fields = {
+        'id': fields.IntegerField(),
+        'uuid': fields.UUIDField(),
+        'from_owner_id': fields.StringField(),
+        'to_owner_id': fields.StringField(),
+        'resource_type': fields.StringField(),
+        'resource_uuid': fields.StringField(),
+        'start_time': fields.DateTimeField(nullable=True),
+        'end_time': fields.DateTimeField(nullable=True),
+        'fulfill_time': fields.DateTimeField(nullable=True),
+        'expire_time': fields.DateTimeField(nullable=True),
+        'status': fields.StringField(),
+    }
+
+    @classmethod
+    def get(cls, owner_change_uuid, context=None):
+        db_owner_change = cls.dbapi.owner_change_get_by_uuid(owner_change_uuid)
+        if db_owner_change:
+            return cls._from_db_object(context, cls(), db_owner_change)
+
+    @classmethod
+    def get_all(cls, filters, context=None):
+        db_owner_changes = cls.dbapi.owner_change_get_all(filters)
+        return cls._from_db_object_list(context, db_owner_changes)
+
+    def create(self, context=None):
```

Comment:
Do we need to check if from_owner_id != to_owner_id?

Reply:
Technically nothing bad would happen, but it's probably worth checking. I've added this check (and a corresponding test) in the API controller.